src/components/Tabs/Annotation/AnnotationDisplayList.tsx

```diff
@@ -60,77 +55,134 @@ export default function AnnotationDisplayList(props: AnnotationDisplayListProps)
   };
 
   // Organize ids by track
-  const trackToIds: Record<string, number[]> = useMemo(() => {
+  const trackToIds: Map<string, number[]> = useMemo(() => {
     if (!props.dataset) {
-      return {};
+      return new Map();
     }
-    const map: Record<string, number[]> = {};
+    const map: Map<string, number[]> = new Map();
     for (const id of props.ids) {
       const trackId: string = props.dataset.getTrackId(id).toString();
-      if (trackId in map) {
-        map[trackId].push(id);
+      if (!map.has(trackId)) {
+        map.set(trackId, [id]);
       } else {
-        map[trackId] = [id];
+        const ids = map.get(trackId);
+        if (ids) {
+          ids.push(id);
+        }
       }
     }
     return map;
   }, [props.dataset, props.ids]);
 
-  const trackIdsSorted = useMemo(() => {
-    return Object.keys(trackToIds)
+  // Track IDs, in order of appearance in the ID list. The track that was last
+  // added will be at the top of the list.
+  const trackIds = useMemo(() => {
+    return Array.from(trackToIds.keys())
       .map((trackId) => parseInt(trackId, 10))
-      .sort((a, b) => a - b);
+      .reverse();
   }, [trackToIds]);
 
   let listContents;
-  if (props.ids.length === 0) {
+  if (props.ids.length === 0 || props.dataset === null) {
     // Show placeholder if there are no elements
     listContents = (
-      // Padding here keeps the icon aligned with the table view
-      <FlexRowAlignCenter style={{ width: "100% ", height: "100px", paddingRight: "24px" }}>
+      <FlexRowAlignCenter style={{ width: "100% ", height: "100px" }}>
         <FlexColumnAlignCenter style={{ margin: "16px 0 10px 0", width: "100%", color: theme.color.text.disabled }}>
           <TagIconSVG style={{ width: "24px", height: "24px", marginBottom: 0 }} />
-          <p>No annotated IDs</p>
+          <p>Labeled tracks will appear here.</p>
         </FlexColumnAlignCenter>
       </FlexRowAlignCenter>
     );
   } else {
     listContents = (
-      <StyledCollapse size="small" ghost>
-        {trackIdsSorted.map((trackId) => {
-          const ids = trackToIds[trackId.toString()];
+      <ul style={{ marginTop: 0 }}>
+        {trackIds.map((trackId) => {
+          const ids = trackToIds.get(trackId.toString())!;
```

Comment:
Looks like this value is only used for its length, and only to create text content in the page. Rather than asserting that the value is present here (which I'm sure it's always expected to be), is it acceptable to handle the case where it's not in that expression? E.g. `ids?.length ?? 0`

Reply:
That makes sense to me!